### PR TITLE
Clarify what happens when limits are violated

### LINKED ISSUE #108: Clarify Baggage Propagation Limits
The [Limits Section](https://w3c.github.io/baggage/#limits) in the Baggage Propagation document specifies the following:

> 1. A platform MUST propagate all list-members up to at least 64 list-members including any list-members added by the platform.
> 2. A platform MUST propagate all list-members including any list-members added by the platform if the resulting baggage-string would be 8192 bytes or less. If the resulting baggage-string would be greater than 8192 bytes, some list-members MAY be dropped until the resulting baggage-string is 8192 characters or less.

It isn't clear to me what is expected the following hypothetical scenario:
- Baggage has 50 list-members
- The baggage-string is 10KB

The first limit specification suggests the entire string must be propagated because it contains less than 64 list-members. OTOH, the second limit specification suggests some list-members MAY be dropped. However, doing so contradicts the first requirement, which is labelled as a MUST.

I don't see any size limit to a list-member the way [it is defined](https://w3c.github.io/baggage/#definition), which implies that the suggested hypotetical scenario is possible.

Pracitcally speaking, it seems to me that if there is a limit on length is provided, the intent is to allow an implemenation to bound the resources it would need to process a baggage string, if necessary. So, this leads me to believe the intent is to allow the size limit to trump the number of list-members limit.

I'm looking for feedback to confirm my thinking on this. Thanks!

### PR DESCRIPTION
**Pull request by dyladan**
Resolves #108

***
<a href="https://pr-preview.s3.amazonaws.com/dyladan/baggage/pull/113.html" title="Last updated on Oct 6, 2022, 7:33 PM UTC (c8ab01a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/baggage/113/326b25b...dyladan:c8ab01a.html" title="Last updated on Oct 6, 2022, 7:33 PM UTC (c8ab01a)">Diff</a>

### CODE REVIEW

**Review by yurishkuro — COMMENTED** [review]
I find the wording "unless N is violated" pretty confusing. I would rather state that there are two size limits, 64/8192, and if the baggage does not satisfy both or either, then the list items MAY be dropped until both limits are satisfied.

File: baggage/HTTP_HEADER_FORMAT.md [hunk 1]
```diff
@@ -77,8 +77,7 @@ If there are multiple `baggage` headers, all limits apply to the combination of
 
 1. A platform MUST propagate all `list-member`s up to _at least_ 64 `list-member`s including any `list-member`s added by the platform.
```
Comment (kalyanaj) [inline baggage/HTTP_HEADER_FORMAT.md:78]:
```suggestion
1. A platform MUST propagate all `list-member`s including any `list-member`s added by the platform whenever both of these conditions are met:

**Condition 1:** The resulting `baggage-string` contains 64 `list-member`s or less.
**Condition 2:** The resulting `baggage-string` is of size 8192 bytes or less. 

2. If the resulting `baggage-string` would contain more than 64 `list-member`s, a platform MAY drop certain `list-member`s at its own discretion until the resulting `baggage-string` contains 64 `list-member`s or less.

3. If the resulting `baggage-string` would contain more than 8192 bytes, a platform MAY drop certain `list-member`s at its own discretion until the resulting `baggage-string` is of size 8192 bytes or less.
```

File: baggage/HTTP_HEADER_FORMAT.md [hunk 2]
```diff
@@ -77,8 +77,7 @@ If there are multiple `baggage` headers, all limits apply to the combination of
...
 2. A platform MUST propagate all `list-member`s including any `list-member`s added by the platform if the resulting `baggage-string` would be 8192 bytes or less. If the resulting `baggage-string` would be greater than 8192 bytes, some `list-member`s MAY be dropped until the resulting `baggage-string` is 8192 characters or less.
```
Comment (kalyanaj) [inline baggage/HTTP_HEADER_FORMAT.md:79]:
```suggestion
```

File: baggage/HTTP_HEADER_FORMAT.md [hunk 3]
```diff
@@ -77,8 +77,7 @@ If there are multiple `baggage` headers, all limits apply to the combination of
...
-
-If a platform does not propagate all `list-member`s, it is left to the implementer to decide which `list-member`s to propagate.
+3. If either (1) or (2) is violated, the platform MAY drop `list-member`s at its own discretion. If a platform does not propagate all `list-member`s, it is left to the implementer to decide which `list-member`s to propagate.
```
Comment (kalyanaj) [inline baggage/HTTP_HEADER_FORMAT.md:80]:
```suggestion
```

**Review by kalyanaj — COMMENTED** [review]
I have a suggestion to simplify this language a bit more...

File: baggage/HTTP_HEADER_FORMAT.md (hunk 1 above)
Comment (yurishkuro) [reply baggage/HTTP_HEADER_FORMAT.md:78]: I like this, but I would not separate 2 and 3, as it's a lot of nearly identical wording just to say that if C1 or C2 or both are violated then drop the items as you see fit.

**Comment by dyladan** [comment]
Thank you @kalyanaj and @yurishkuro for your suggestions. I simplified the wording PTAL.

**Review by yurishkuro — APPROVED** [review]

**Review by kalyanaj — APPROVED** [review]

**Pull request merged**